src/nvim/api/buffer.c

```diff
@@ -1932,6 +1944,17 @@ Object nvim_buf_call(Buffer buffer, LuaRef fun, Error *err)
   Array args = ARRAY_DICT_INIT;
   Object res = nlua_call_ref(fun, NULL, args, true, err);
 
+  bufref_T bufref = bufref_from(buf);
```

Comment:
This seems wrong. the return of a function call should not be an interesting time where things implicitly happen. Either this checking should be done at the "inner" modifying api call (as is already done in some places) or alternatively when we return to something interesting, like the event loop of the active mode.

Reply:
Oh ok I see what you mean. But given that the autocommands check is only done for current buffer/window, doesn't that mean that autocommands won't dispatch until the buffer/window is made current? Cuz for neovim API, we do dispatch autocommands when stuff happens, but the rest of the vim codebase assumes that stuff will be checked in the next iteration of the event loop, so we might get the same problem as before that autocommands aren't dispatched when they should. Or do you think we should just do the inner check, and progressively fix the vim parts of the codebase? Or maybe add a queue of stuff to check for the next iteration of the event loop that would check autocommands like this? Eg

```c
// here
queue_autocmd_check(bufferHandle, EVENT_TEXTCHANGED);
// ...and the other checks

// ... then inside the event loop iteration
if (!autocmd_check_queue.isEmpty())
  do_checks(autocmd_check_queue);

```